modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js

```diff
@@ -12,239 +12,145 @@
-			jest.runAllTimers();
-		});
-
 		expect(container.querySelector('.form-text')).toBeTruthy();
 	});
 
 	it('has a label', () => {
-		const {container} = render(
-			<DatePickerWithProvider
-				{...defaultDatePickerConfig}
-				label="label"
-			/>
-		);
-
-		act(() => {
-			jest.runAllTimers();
-		});
+		const {container} = render(<DatePicker label="label" />);
 
 		expect(container.querySelector('.ddm-label')).toBeTruthy();
 	});
 
 	it('has a predefinedValue', () => {
 		const {container} = render(
-			<DatePickerWithProvider
-				{...defaultDatePickerConfig}
-				predefinedValue="06/02/2020"
+			<DatePicker
+				label="date"
+				locale="en_US"
+				localizedValue={{}}
+				name="dateField"
+				predefinedValue="2020-06-02"
 			/>
 		);
 
-		act(() => {
-			jest.runAllTimers();
-		});
+		const inputDatePicker = container
+			.getElementsByClassName('form-control')
+			.item(0);
 
-		expect(container).toMatchSnapshot();
+		expect(inputDatePicker).toHaveValue('06/02/2020');
 	});
 
-	it('expands the datepicker when clicking the calendar icon', async () => {
-		const {container} = render(
-			<DatePickerWithProvider {...defaultDatePickerConfig} />
-		);
+	it('expands the datepicker when clicking the calendar icon', () => {
+		const {container} = render(<DatePicker />);
 
-		userEvent.click(
-			container.querySelector('.date-picker-dropdown-toggle')
+		const btnDropdown = container.querySelector(
+			'.date-picker-dropdown-toggle'
 		);
 
-		act(() => {
-			jest.runAllTimers();
-		});
+		userEvent.click(btnDropdown);
 
-		await wait(() =>
-			expect(
-				document.body.querySelector('.date-picker-dropdown-menu.show')
-			).toBeTruthy()
+		const dropdownDatePicker = document.body.querySelector(
+			'.date-picker-dropdown-menu.show'
 		);
+
+		expect(dropdownDatePicker).toBeTruthy();
 	});
 
-	it('fills the input with the current date selected on Date Picker', async () => {
-		const handleFieldEdited = jest.fn();
+	it('fills the input with the current date selected on Date Picker', () => {
+		const onChange = jest.fn();
 
 		const {container, getAllByDisplayValue, getByLabelText} = render(
-			<DatePickerWithProvider
-				{...defaultDatePickerConfig}
-				onChange={handleFieldEdited}
-			/>
+			<DatePicker locale="en_US" name="dateField" onChange={onChange} />
 		);
 
-		userEvent.click(
-			container.querySelector('.date-picker-dropdown-toggle')
+		const btnDropdown = container.querySelector(
+			'.date-picker-dropdown-toggle'
 		);
 
-		act(() => {
-			jest.runAllTimers();
-		});
-
+		userEvent.click(btnDropdown);
 		userEvent.click(getByLabelText('Select current date'));
 
-		act(() => {
-			jest.runAllTimers();
-		});
+		const expectedDate = moment().format('MM/DD/YYYY');
 
-		await wait(() =>
-			expect(
-				getAllByDisplayValue(moment().format('MM/DD/YYYY'))
-			).toHaveLength(2)
-		);
-
-		expect(handleFieldEdited).toHaveBeenCalled();
+		expect(onChange).toHaveBeenCalled();
+		expect(getAllByDisplayValue(expectedDate)).toHaveLength(2);
```

Comment:
- Try to query element by role or label to ensure the component's accessibility, according to the [query priority](https://testing-library.com/docs/queries/about/#priority) 
- Use [tohavevalue](https://github.com/testing-library/jest-dom#tohavevalue) to check the value of an `input` element:
```suggestion
		expect(getByRole('textbox')).tohavevalue(displayDate);
```